src/Core/Services/VaultTimeoutService.cs

```diff
@@ -19,7 +19,7 @@ public class VaultTimeoutService : IVaultTimeoutService
         private readonly ITokenService _tokenService;
         private readonly IPolicyService _policyService;
         private readonly IKeyConnectorService _keyConnectorService;
-        private readonly Action<bool> _lockedCallback;
+        private readonly Func<Tuple<string, bool>, Task> _lockedCallback;
```

Comment:
I think that named tuples are better because one can know what the parameters are inline:
` private readonly Func<(string userId, bool userInitiated), Task> _lockedCallback`

Reply:
Fixed in #1830